### PR TITLE
Development

### PR DESCRIPTION
**Pull request by sehalondas**

### CODE REVIEW

File: api/orders.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+const router = express.Router();
+
+router.get("/orders", async (req, res, next) => {
+    try {
+    }   catch (error) {
+        next(error);
+    }
+});
+
+router.get("/", async (req, res, next) => {
```
Comment (parisyee) [inline api/orders.js:14]: I notice that these routes are all prefixed with `/orders`, but does that mean this from api/index.js doesn't tack the `/orders` on to the front of the path?
```
const ordersRouter = require("./orders");
router.use("/orders", ordersRouter);
```
It's not clear to me why this route would be different than the others in this file.

File: api/orders.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    try{
+        const order = await getOrderById(orderId);
+        if(order) {
+            res.send(order);
+        }   else {
+            res.send({
+                error: "ERROR",
+                title: "OrderNotFound",
+                message: `order ${orderId} not found.`
+            });
```
Comment (parisyee) [inline api/orders.js]: You'll want the response to include a 404 code for not found

File: api/orders.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        }
+    }   catch (error) {
+        next(error);
+    }
+});
+
+router.get("/order/:userId", async (req, res, next) => {
```
Comment (parisyee) [inline api/orders.js]: why is `order` singular? What if a user had multiple?

File: api/orders.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+            orderStatus: orderStatus,
+            quantity: quantity
+        });
+        res.send(updatedOrder);
+    }   catch (error) {
+        next(error);
+    }
+});
+
+router.patch("/orders/:productId", async (req, res, next) => {
```
Comment (parisyee) [inline api/orders.js]: Why are you putting this behavior in the `orders` namespace? Seems like a product concern, no?

File: api/products.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    const product = await getProductById(productId);
+
+    if (product) {
+      res.send(product);
+    } else {
+      res.send({
+        error: "ERROR",
+        message: `product ${productId} not found`,
+        title: "productNotFound",
+      });
```
Comment (parisyee) [inline api/products.js]: Same thing for these... send response codes.

File: db/models/products.js [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+  }
+}
+
+async function getProductsBySearch({ searchTerm }) {
+  try {
+    const { rows: products } = await client.query(
+      `
+      SELECT products.*
+      FROM products
+      WHERE title
```
Comment (parisyee) [inline db/models/products.js:104]: don't you want `searchTerm`?

**Review by parisyee — CHANGES_REQUESTED** [review]
Not too shabby! Just curious about those few comments. But in general, this looks great!

File: README.md [hunk 7]
```diff
@@ -1,143 +1,150 @@
-# Let's Build a Full Stack Application
+# Project Goals
```
Comment (leafoflegend) [inline README.md:1]: Love the use of markdown!

File: api/orders.js (hunk 3 above)
Comment (leafoflegend) [reply api/orders.js]: I might even argue this feels slightly inverted to me as well. Like, orders belong to a user, not users belong to orders. I might do this route over on the user side.

File: api/utils.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
+const requireAuthentication = (req, res, next) => {
+    const isAuthenticated = req.headers.authorization;
+  
+    if (isAuthenticated) {
+      next();
```
Comment (leafoflegend) [inline api/utils.js]: This isn't security, this is a stub. I hope theres intentions to actually inspect this header and validate its something meaningful? As of right now, every route is a security vulnerability.

File: db/seedData.js [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+        description TEXT NOT NULL,
+        price INTEGER NOT NULL,
+        quantity INTEGER NOT NULL,
+        category VARCHAR(255) NOT NULL,
+        photo BYTEA
+      );
+      CREATE TABLE orders (
+        id SERIAL PRIMARY KEY,
+        "userId" INTEGER REFERENCES users(id),        
+        "productId" INTEGER REFERENCES products(id),
```
Comment (leafoflegend) [inline db/seedData.js]: Don't many products belong to a single order? This infers only one does.

File: src/components/App.jsx [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+  Account,
+  Wishlist,
+  Orders,
+  Register,
+  Login,
+  Search,
+  Categories,
+} from "../components/index.js";
+
+// This is the Mother of all components. This is what will house all of the other components to render on screen.
```
Comment (leafoflegend) [inline src/components/App.jsx:32]: 👍

File: src/components/Body/AllProducts.jsx [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+  const filterProducts = () => {
+    const filtered = products.filter((product) => {
+      const lowerCaseQuery = searchTerm.toLowerCase();
+      return (
+        product.title.toLowerCase().includes(lowerCaseQuery) ||
+        product.description.toLowerCase().includes(lowerCaseQuery)
+      );
+    });
+    setFilteredProducts(filtered);
+  };
```
Comment (leafoflegend) [inline src/components/Body/AllProducts.jsx]: Nice!

File: src/components/Header/Header.jsx [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+                  Register
+                </NavDropdown.Item>
+              </>
+            )}
+            {newUser === false && (
+              <>
+                <NavDropdown.Item href="/" onClick={handleLogout}>
+                  Logout
+                </NavDropdown.Item>
+              </>
```
Comment (leafoflegend) [inline src/components/Header/Header.jsx]: Do you need `href`s on these? Not sure they're totally valid.

**Review by leafoflegend — CHANGES_REQUESTED** [review]
I like how far the front end has come along! The database has some pretty serious issues relating to the models - you may need to rethink some of whats going on there. Need a pretty big push on the backend here as well as the frontend.

Theres also not even a vague concept of security on the backend. Absolutely everything is vulnerable.

**Pull request merged**